### PR TITLE
Issue #6207: Added Xpath IT regression test for StaticVariableName

### PR DESCRIPTION
**Pull request by aayushRedHat**
Related: https://github.com/checkstyle/checkstyle/issues/6207

Diff Regression config: https://gist.githubusercontent.com/aayushRedHat/849313f94f6df07629e364070d5ed19a/raw/3fc325f6c384a7ca918dc4a5875fa26cc1444679/gistfile1.txt

Diff Regression projects: https://gist.githubusercontent.com/aayushRedHat/085b2f4c08f7e2859c92e0d2cb2144c9/raw/358a901648397fd9f66978d4ce51636694546177/gistfile1.txt

### CODE REVIEW

**Comment by aayushRedHat** [comment]
Github, generate report

**Review by nrmancuso — CHANGES_REQUESTED** [review]
https://github.com/checkstyle/checkstyle/pull/14074#discussion_r1410096873

**Comment by aayushRedHat** [comment]
> #14074 (comment)

@nrmancuso Added test cases

File: src/it/resources/org/checkstyle/suppressionxpathfilter/staticvariablename/SuppressionXpathRegressionStaticVariableName2.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package org.checkstyle.suppressionxpathfilter.staticvariablename;
+
+public class SuppressionXpathRegressionStaticVariableName2 {
+
+     public int num1; // OK
+     protected static int NUM2; // warn
```
Comment (romani) [inline src/it/resources/org/checkstyle/suppressionxpathfilter/staticvariablename/SuppressionXpathRegressionStaticVariableName2.java:6]: It is almost same as input1, please wrap this field in inners class to show violation on field of inner class.

**Review by romani — CHANGES_REQUESTED** [review]
Small improvement

**Comment by nrmancuso** [comment]
> > #14074 (comment)
> 
> @nrmancuso Added test cases

Not done. We only need two test inputs, one with what is is the first test, one with what I suggest above (static field in local class).

**Comment by romani** [comment]
@aayushRedHat, ping

**Comment by aayushRedHat** [comment]
> @aayushRedHat, ping

Done @romani Sorry for being late

**Review by nrmancuso — APPROVED** [review]

**Comment by aayushRedHat** [comment]
> from issue:
> 
> > There should be at least 3 test methods for different structure of code.
> 
> @aayushRedHat, please add one more .

Sure @romani

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionStaticVariableNameTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        );
+        runVerifications(moduleConfig, fileToProcess, expectedViolation,
+                expectedXpathQueries);
+    }
+
+    @Test
+    public void test2() throws Exception {
+        final File fileToProcess =
+                new File(getNonCompilablePath(
+                        "SuppressionXpathRegressionStaticVariableName1.java"));
```
Comment (romani) [inline src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionStaticVariableNameTest.java:70]: Use second input file

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionStaticVariableNameTest.java (hunk 2 above)
Comment (aayushRedHat) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionStaticVariableNameTest.java]: Done @romani

**Review by romani — APPROVED** [review]
Ok to merge, thanks a lot !!

**Pull request merged**